operator/pkg/helmreconciler/rendering.go

```diff
@@ -175,19 +189,77 @@ func unmarshalAndValidateIOPSpec(iopsYAML string) (*v1alpha1.IstioOperatorSpec,
 func (h *HelmReconciler) ProcessManifest(manifest manifest.Manifest) (int, error) {
 	var errs []error
 	log.Infof("Processing resources from manifest: %s", manifest.Name)
-	objects, err := object.ParseK8sObjectsFromYAMLManifest(manifest.Content)
+	allObjects, err := object.ParseK8sObjectsFromYAMLManifest(manifest.Content)
 	if err != nil {
 		return 0, err
 	}
-	for _, obj := range objects {
+
+	name := h.instance.Name
+
+	objectCachesMu.Lock()
+
+	// Create and/or get the cache corresponding to the CR name we're processing. Per name partitioning is required to
+	// prune the cache to remove any objects not in the manifest generated for a given CR.
+	if objectCaches[name] == nil {
+		objectCaches[name] = &ObjectCache{
+			cache: make(map[string]*object.K8sObject),
+			mu:    &sync.RWMutex{},
+		}
+	}
+	objectCache := objectCaches[name]
+
```

Comment:
`objectCachesMu.Unlock()` here instead of later?

Reply:
Done.